### PR TITLE
Blog app: add devise

### PR DESCRIPTION
**Pull request by Nahid1911**
### Project requirements
- Remove `current_user` method in `ApplicationController`, devise will provide us with one.
- [Install and setup devise](https://github.com/heartcombo/devise#getting-started).
  - Can register a new user.
  - User logs in with a combination of email and password.
  - Hashed password should be stored in the database.
  - Ask for confirmation of email.
  - Can reset password.

For more details please click [HERE](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_devise.md)

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,18 +1,27 @@
...
+  before_action :update_allowed_parameters, if: :devise_controller?
 
   private
 
-  def set_current_user
-    user_id = params[:user_id]
-    @current_user = User.find_by(id: user_id)
+  def current_user
+    @current_user ||= warden.user
   end
```
Comment (t-yanick) [inline app/controllers/application_controller.rb]: - [ ] Great job working on this milestone. As per project requirements, the current_user method should be removed as devise provides a way of doing this.
Screen Shot 2023-09-07 at 10 20 34 AM

File: app/models/user.rb [hunk 2]
```diff
@@ -1,4 +1,8 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (t-yanick) [inline app/models/user.rb]: - [ ] You have a view for confirmation at [app/views/devise/confirmations/new.html.erb](https://github.com/Nahid1911/blog-app-ruby-on-rails/pull/10/files#diff-eb9d2217df62505f3e24b1eff9e71afc7559219ebd549d7fbcb902bb245f2582) , but you need to set the module `confirmable` in `user.rb`

**Review by t-yanick — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED

Hi @Nahid1911 , 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Collaborative work 👍 Commits from both team members
:heavy_check_mark: Effective use of Devise for authentication
:heavy_check_mark: Use of correct Gitflow
:heavy_check_mark: Professional documentation


There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 2 above)
Comment (Nahid1911) [reply app/models/user.rb]: Resolved

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (Nahid1911) [reply app/controllers/application_controller.rb]: resolved

**Review by mckent05 — APPROVED** [review]
Hi @Nahid191, 

# STATUS: APPROVED ✔️ ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations and well done! 🎉 
approved


## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👏 👏 
- Good job adding devise to your application 👏 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**